src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/versionMatcher/versionObjectDpkg.hpp

```diff
@@ -0,0 +1,158 @@
+/*
+ * Wazuh Vulnerability scanner - Database Feed Manager
+ * Copyright (C) 2015, Wazuh Inc.
+ * December 06, 2023.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#ifndef _VERSION_OBJECT_DPKG_HPP
+#define _VERSION_OBJECT_DPKG_HPP
+
+#include "iVersionObjectInterface.hpp"
+#include <iostream>
+#include <memory>
+#include <regex>
+#include <string>
+
+/**
+ * @brief dpkgVersion data struct.
+ *
+ */
+struct dpkgVersion
+{
+    uint32_t epoch;       ///< Epoch.
+    std::string version;  ///< Version.
+    std::string revision; ///< Revision.
+};
+
+/**
+ * @brief VersionObjectDpkg class.
+ *
+ */
+class VersionObjectDpkg final : public IVersionObject
+{
+private:
+    static std::regex m_parserRegex;
+    uint32_t m_epoch;
+    std::string m_version;
+    std::string m_revision;
+
+public:
+    /**
+     * @brief Static method to match a version string to a dpkgVersion object.
+     *
+     * @param version version string to match.
+     * @param output dpkgVersion object to store the result.
+     * @return true/false according to match condition.
+     */
+    static bool match(const std::string& version, dpkgVersion& output)
+    {
+        std::smatch parserMatches;
+        if ((std::regex_match(version, parserMatches, m_parserRegex) == false) || (parserMatches.size() != 4))
+        {
+            return false;
+        }
+
+        try
+        {
+            output.epoch = static_cast<uint32_t>(std::stoul(parserMatches.str(1)));
```

Comment:
This try/catch block could be replaced with something like this:
`output.epoch = parserMatches.str(1).empty() ? 0 : static_cast<uint32_t>(std::stoul(parserMatches.str(1)));`